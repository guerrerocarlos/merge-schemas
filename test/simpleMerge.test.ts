import mergeSchema from "../src";
import {
  GraphQLSchema,
  GraphQLObjectType,
  GraphQLString,
  GraphQLID
} from "graphql";

test("Basic GraphQL Schema Merge", () => {
  const customSchema = new GraphQLSchema({
    query: new GraphQLObjectType({
      name: "RootQueryType",
      fields: {
        health: {
          type: GraphQLString,
          resolve() {
            return "OK";
          }
        }
      }
    }),
    mutation: new GraphQLObjectType({
      name: "RootMutationType",
      fields: {
        sendMessage: {
          type: new GraphQLObjectType({
            name: "sendMessageMutation",
            fields: {
              text: {
                type: GraphQLString
              },
              id: {
                type: GraphQLID
              },
              type: {
                type: GraphQLString
              }
            }
          }),
          args: {
            text: {
              type: GraphQLString
            },
            type: {
              type: GraphQLString
            }
          },
          description: "Send a new message to all",
          async resolve(_, { text, type }) {
            console.log("{ text, type }", { text, type });
            const payload = { id: String(Math.random() * 1000), text, type };
            console.log("send new message!", payload);
            return payload; // { success: true };
          }
        }
      }
    })
  });

  const resultSchema = mergeSchema({ schemas: [customSchema] })

  expect(resultSchema).toMatchSnapshot()

});
