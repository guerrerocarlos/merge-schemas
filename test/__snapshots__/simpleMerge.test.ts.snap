// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic GraphQL Schema Merge 1`] = `
GraphQLSchema {
  "__allowedLegacyNames": Array [],
  "__validationErrors": undefined,
  "_directives": Array [
    "@include",
    "@skip",
    "@deprecated",
  ],
  "_implementations": Object {},
  "_mutationType": "RootMutationType",
  "_possibleTypeMap": Object {},
  "_queryType": "RootQueryType",
  "_subscriptionType": "RootSubscriptionType",
  "_typeMap": Object {
    "Boolean": "Boolean",
    "RootMutationType": "RootMutationType",
    "RootQueryType": "RootQueryType",
    "RootSubscriptionType": "RootSubscriptionType",
    "String": "String",
    "__Directive": "__Directive",
    "__DirectiveLocation": "__DirectiveLocation",
    "__EnumValue": "__EnumValue",
    "__Field": "__Field",
    "__InputValue": "__InputValue",
    "__Schema": "__Schema",
    "__Type": "__Type",
    "__TypeKind": "__TypeKind",
  },
  "astNode": undefined,
  "extensionASTNodes": undefined,
  "extensions": undefined,
}
`;
